service/history/api/recordactivitytaskstarted/api.go

```diff
@@ -101,6 +107,61 @@ func Invoke(
 	return response, err
 }
 
+func handleChasmActivityStarted(
+	ctx context.Context,
+	activityRefProto *persistencespb.ChasmComponentRef,
+	request *historyservice.RecordActivityTaskStartedRequest,
+) (*historyservice.RecordActivityTaskStartedResponse, error) {
+	activityRef := chasm.ProtoRefToComponentRef(activityRefProto)
+
+	updatedActivity, err := activity.HandleRecordActivityTaskStarted(
+		ctx, activityRef, request.GetVersionDirective(), request.GetPollRequest().GetIdentity())
+	if err != nil {
+		return nil, err
+	}
+
+	requestData, err := activity.GetActivityRequestData(ctx, activityRef)
+	if err != nil {
+		return nil, err
+	}
+
+	attempt, err := activity.GetActivityAttempt(ctx, activityRef)
+	if err != nil {
+		return nil, err
+	}
+
+	lastHeartbeat, err := activity.GetActivityLastHeartbeat(ctx, activityRef)
+	if err != nil {
+		return nil, err
+	}
+
+	options := updatedActivity.GetActivityOptions()
```

Comment:
There's no point in locking the activity 4 times, create one function that is executed once under a lock instead and returns `RecordActivityTaskStartedResponse`.

Also, we mentioned writing the code in a way that would work for a workflow CHASM activity. Please make sure we use the `Store` interface (we should rename it but the concept should still be there).